tests/jax/lax/test_dispatch_combine.py

```diff
@@ -0,0 +1,263 @@
+###############################################################################
+# Copyright (c) 2025, Advanced Micro Devices, Inc. All rights reserved.
+#
+# See LICENSE for license information.
+###############################################################################
+
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+import pytest
+from jax.sharding import PartitionSpec
+
+from primus_turbo.jax.lax.moe import get_dispatch_config, moe_combine, moe_dispatch
+from primus_turbo.jax.primitive.moe.moe_dispatch import moe_dispatch_p
+
+key = jax.random.PRNGKey(123)
+
+num_ranks = jax.device_count()
+
+mesh = jax.make_mesh((num_ranks,), ("x",), axis_types=(jax.sharding.AxisType.Explicit,))
+
+
+def _generate(num_tokens, hidden, num_topk, num_experts):
+    x = jnp.ones((num_ranks, num_tokens, hidden), dtype=jnp.bfloat16)
+    scores = jnp.abs(jax.random.normal(key, (num_ranks, num_tokens, num_experts), dtype=jnp.float32)) + 1
+
+    topk_weights = jnp.ones((num_ranks, num_tokens, num_topk), dtype=jnp.float32)
+
+    for i in range(num_ranks):
+        x = x.at[i].set(i)
+        topk_weights = topk_weights.at[i].set(i)
+    return (
+        jnp.reshape(x, (num_ranks * num_tokens, hidden)),
+        jnp.reshape(scores, (num_ranks * num_tokens, num_experts)),
+        jnp.reshape(topk_weights, (num_ranks * num_tokens, num_topk)),
+    )
```

Comment:
Can we add a comment on how the different items in the tuples correspond to B, M, N, K?